src/Message.php

```diff
@@ -0,0 +1,289 @@
+    }
+
+    public function getCc(): array|string
+    {
+        return $this->cc;
+    }
+
+    public function withCc(array|string $cc): MessageInterface
+    {
+        $new = clone $this;
+        $new->cc = $cc;
+        return $new;
+    }
+
+    public function getBcc(): array|string
+    {
+        return $this->bcc;
+    }
+
+    public function withBcc(array|string $bcc): MessageInterface
+    {
+        $new = clone $this;
+        $new->bcc = $bcc;
+        return $new;
+    }
+
+    public function getSubject(): string
+    {
+        return $this->subject;
+    }
+
+    public function withSubject(string $subject): MessageInterface
+    {
+        $new = clone $this;
+        $new->subject = $subject;
+        return $new;
+    }
+
+    public function getDate(): ?DateTimeImmutable
+    {
+        return $this->date;
+    }
+
+    public function withDate(DateTimeInterface $date): MessageInterface
+    {
+        $new = clone $this;
+        $new->date = $date instanceof DateTimeImmutable ? $date : DateTimeImmutable::createFromInterface($date);
+        return $new;
+    }
+
+    public function getPriority(): int
+    {
+        return $this->priority;
+    }
+
+    public function withPriority(int $priority): MessageInterface
+    {
+        $new = clone $this;
+        $new->priority = $priority;
+        return $new;
+    }
+
+    public function getReturnPath(): string
+    {
+        return $this->returnPath;
+    }
+
+    public function withReturnPath(string $address): MessageInterface
+    {
+        $new = clone $this;
+        $new->returnPath = $address;
+        return $new;
+    }
+
+    public function getSender(): string
+    {
+        return $this->sender;
+    }
+
+    public function withSender(string $address): MessageInterface
+    {
+        $new = clone $this;
+        $new->sender = $address;
+        return $new;
+    }
+
+    public function getHtmlBody(): string
+    {
+        return $this->htmlBody;
+    }
+
+    public function withHtmlBody(string $html): MessageInterface
+    {
+        $new = clone $this;
+        $new->htmlBody = $html;
+        return $new;
+    }
+
+    public function getTextBody(): string
+    {
+        return $this->textBody;
+    }
+
+    public function withTextBody(string $text): MessageInterface
+    {
+        $new = clone $this;
+        $new->textBody = $text;
+        return $new;
+    }
+
+    public function getAttachments(): array
+    {
+        return $this->attachments;
+    }
+
+    public function withAttached(File $file): MessageInterface
+    {
+        $new = clone $this;
+        $new->attachments[] = $file;
+        return $new;
+    }
+
+    public function getEmbeddings(): array
+    {
+        return $this->embeddings;
+    }
+
+    public function withEmbedded(File $file): MessageInterface
+    {
+        $new = clone $this;
+        $new->embeddings[] = $file;
+        return $new;
+    }
+
+    public function getHeader(string $name): array
```

Comment:
Maybe add `withAttachments()` and `withEmbeddings()` methods  to add multiple files at once in a more convenient way, similar to `withHeaders()` method?

Reply:
Allowed to pass several files to `withAttachment()`/`withEmbedding()` and add new methods `withAddedAttachment()`/`withAddedEmbedding()`.